sdk/spring/azure-spring-integration-servicebus/src/main/java/com/azure/spring/integration/servicebus/converter/ServiceBusMessageConverter.java

```diff
@@ -73,32 +77,56 @@ protected ServiceBusMessage fromByte(byte[] payload) {
     @Override
     protected void setCustomHeaders(MessageHeaders headers, ServiceBusMessage message) {
 
+        Map<String, Object> copyHeaders = new HashMap<String, Object>();
+        copyHeaders.putAll(headers);
+
         // Spring MessageHeaders
-        getStringHeader(headers, MessageHeaders.ID).ifPresent(message::setMessageId);
-        getStringHeader(headers, MessageHeaders.CONTENT_TYPE).ifPresent(message::setContentType);
-        getStringHeader(headers, MessageHeaders.REPLY_CHANNEL).ifPresent(message::setReplyTo);
+        getStringHeader(headers, copyHeaders, MessageHeaders.ID).ifPresent(message::setMessageId);
+        getStringHeader(headers, copyHeaders, MessageHeaders.CONTENT_TYPE).ifPresent(message::setContentType);
+        getStringHeader(headers, copyHeaders, MessageHeaders.REPLY_CHANNEL).ifPresent(message::setReplyTo);
```

Comment:
We'd better use another method (name) for this, getStringHeader doesn't fit anymore.

Reply:
I change it as getHeaderAsStringAndRemove now, what do you think?